instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/org/springframework/web/servlet/OpenTelemetryHandlerMappingFilter.java

```diff
@@ -72,22 +84,61 @@ public void init(FilterConfig filterConfig) {}
   public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain)
       throws ServletException, IOException {
 
+    ContentCachingResponseWrapper wrapper = new ContentCachingResponseWrapper((HttpServletResponse) response);
```

Comment:
Don't we want to protect this cast? Is it 100% safe?

Reply:
which one? the httpservletResponse ? They are doing it in many more other places

Reply:
ok

Reply:
I moved it after the validation, you'll see in a bit